File: src/core/python/core/model/eventType.py
from enum import Enum


class EventType(Enum):
    """
    Enumeration of all possible event types.
    Arrival and departure are the classical event types, fix is for fixed events, i.e., when parts of a timetable should
    be fixed for the optimization.
    """
    ARRIVAL = "\"arrival\""
    DEPARTURE = "\"departure\""
    FIX = "\"fix\""

    def __init__(self, representation: str):
        self.representation = representation

    def __str__(self) -> str:
        return self.representation

File: src/core/java/src/net/lintim/model/EventType.java
package net.lintim.model;

/**
 * Enumeration of all possible event types.
 * Arrival and departure are the classical event types, fix is for fixed events, i.e., when parts of a timetable should
 * be fixed for the optimization.
 */
public enum EventType {
    ARRIVAL("\"arrival\""),
    DEPARTURE("\"departure\""),
    FIX("\"fix\"");

    private String representation;

    EventType(String representation){
        this.representation = representation;
    }

    public String toString(){
        return representation;
    }
}
